packages/vulcan-forms/lib/components/Form.jsx

```diff
@@ -560,18 +561,29 @@ class SmartForm extends Component {
 
   // TODO: only need to check nextProps.prefilledProps?
   // TODO: see https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html
-  
+
   */
   UNSAFE_componentWillReceiveProps(nextProps) {
     if (!isEqual(this.props, nextProps)) {
       this.setState(getInitialStateFromProps(nextProps));
     }
   }
+  componentWillReceiveProps(nextProps) {
```

Comment:
We probably don't want to use both `UNSAFE_componentWillReceiveProps` and `componentWillReceiveProps`?

Also wouldn't changing the schema already be covered by `!isEqual(this.props, nextProps)`?

Reply:
Wow, I somehow missed the big deprecation of lifecycle methods in React v16.3 😊 
Then I totally misread the code, I thought it's a creative way of disabling the method call, due to some reasons regarding the whole state recalculation. Of course deep comparing whole props is covering the schema case. I've removed the commit altogether

Reply:
However, note that this code does not work in the current `master` of Vulca-Starter, because both `react` and `react-dom` need to be at version ^16.3 (the reason why the state wasn't updated in my case: react-dom v16.2)
I opened a [pull request](https://github.com/VulcanJS/Vulcan-Starter/pull/96) in that repo